File: typescript/src/resistor-color-duo.ts
export const COLORS = [
      'black',
      'brown',
      'red',
      'orange',
      'yellow',
      'green',
      'blue',
      'violet',
      'grey',
      'white',
    ]

export const colorCode = (color:string) => {
  return COLORS.indexOf(color)
}

export function decodedValue([tens, units]: [string, string]) {
  return 10 * colorCode(tens) + colorCode(units)
}

File: java/src/main/java/ResistorColorDuo.java
import java.util.Arrays;


class ResistorColorDuo {
    int value(final String[] colors) {
        return Arrays.stream(colors)
                     .limit(2)
                     .map(this::colorCode)
                     .reduce((a, b) -> 10 * a + b)
                     .orElse(-1);
    }

    private int colorCode(final String color) {
        return ResistorBand.valueOf(color)
                           .ordinal();
    }
}

enum ResistorBand {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white;
}

